core/domain/learner_playlist_services_test.py

```diff
@@ -179,6 +180,46 @@ def test_mark_exploration_to_be_played_later(self):
             self._get_all_learner_playlist_exp_ids(
                 self.user_id), exp_ids)
 
+        # Empty the learner playlist again.
+        for exp_id in exp_ids:
+            learner_playlist_services.remove_exploration_from_learner_playlist(
+                self.user_id, exp_id)
+        self.assertEqual(
+            self._get_all_learner_playlist_exp_ids(
+                self.user_id), [])
+
+        # Subscribe to exploration.
+        subscription_services.subscribe_to_exploration(
+            self.user_id, 'SUBSCRIBED_ID')
+
+        # Now if we try to add the same exploration, it shouldn't be added.
+        learner_progress_services.add_exp_to_learner_playlist(
+            self.user_id, 'SUBSCRIBED_ID')
+        self.assertEqual(
+            self._get_all_learner_playlist_exp_ids(
+                self.user_id), [])
+
+        # Add MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT explorations at given
+        # positions.
+        for i in range(0, MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT):
+            learner_progress_services.add_exp_to_learner_playlist(
+                self.user_id, exp_ids[i], position_to_be_inserted=i)
+        self.assertEqual(
+            self._get_all_learner_playlist_exp_ids(
+                self.user_id), exp_ids)
+
+        # Now if we try to add another explorations at the end of the list, 
```

Comment:
This sounds wrong.... I think it should remain as "exploration" (singular).

Reply:
That's true. Thanks!